code-block/src/components/CodeEditor/CodeEditorHeader.tsx

```diff
@@ -13,58 +13,97 @@ import { integerFromString } from '../../utils'
 export const CodeEditorHeader: FunctionComponent<{
   title: string | undefined
   onTitleChange: (title: string | undefined) => void
+  language: string | undefined
+  onLanguageChange: (language: string | undefined) => void
   lineNumberStart: number | undefined
   onLineNumberStartChange: (lineNumberStart: number | undefined) => void
+  enableTitle: boolean
+  enableLanguage: boolean
+  enableLineNumberStart: boolean
 }> = (props) => {
+  const { enableTitle, enableLanguage, enableLineNumberStart } = props
   const handleTitleChange: ChangeEventHandler<HTMLInputElement> = (e) => {
     const { value } = e.currentTarget
     props.onTitleChange(value === '' ? undefined : value)
   }
 
+  const handleLanguageChange: ChangeEventHandler<HTMLInputElement> = (e) => {
+    const { value } = e.currentTarget
+    props.onLanguageChange(value === '' ? undefined : value)
+  }
+
   const handleLineNumberOffsetChange: ChangeEventHandler<HTMLInputElement> = (
     e,
   ) => {
     const { value } = e.currentTarget
     props.onLineNumberStartChange(integerFromString(value))
   }
 
-  return (
+  return enableTitle || enableLanguage || enableLineNumberStart ? (
```

Comment:
Just a suggestion, we could return <></> or null earlier such as if(!enableTitle && !enableLanguage && !enableLineNumberStart){
return null
}

I am only suggesting this because the jsx has some conditionals already so it could simplify it a bit. It is not necessary tho. Let me know what you think.

Reply:
Thanks for the suggestion, I did decide to return an empty fragment (not `null`, because I prefer `undefined` everywhere for consistency)